### PR TITLE
[TASK] Always use the strict version of `in_array` & friends

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

**Comment by oliverklee** [comment]
We'll need some casts first (in a separate PR).

File: src/RuleSet/DeclarationBlock.php [hunk 1]
```diff
@@ -338,20 +338,21 @@ public function expandFontShorthand(): void
...
                 $aFontProperties['font-style'] = $mValue;
             } elseif ($mValue == 'small-caps') {
                 $aFontProperties['font-variant'] = $mValue;
             } elseif (
-                in_array($mValue, ['bold', 'bolder', 'lighter'])
-                || ($mValue instanceof Size
-                    && in_array($mValue->getSize(), range(100, 900, 100)))
+                in_array($mValue, ['bold', 'bolder', 'lighter'], true)
+                // Note: font weights should be integers, not floats. For now, we'll keep using floats. We will
+                // drill down on this and fix it in  https://github.com/MyIntervals/PHP-CSS-Parser/issues/612.
```
Comment (JakeQZ) [inline src/RuleSet/DeclarationBlock.php:354]: This comment needs changing (or removing), now that we have closed #612 on the basis that CSS treats all numerical values as a `number`, regardless of whether they may be floating-point.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
The comment re `font-weight` needs changing or removing.  Otherwise LGTM.

File: src/RuleSet/DeclarationBlock.php (hunk 1 above)
Comment (oliverklee) [reply src/RuleSet/DeclarationBlock.php]: Done.

**Review by JakeQZ — APPROVED** [review]
LG now.  (Though I didn't notice any 'friends'.)

**Pull request merged**